src/utility/HCISharedMemTransport.cpp

```diff
@@ -0,0 +1,837 @@
+    {
+        SHCI_C2_Ble_Init_Cmd_Packet_t ble_init_cmd_packet = {
+            0, 0, 0,                            /**< Header unused */
+            0,                                  /** pBleBufferAddress not used */
+            0,                                  /** BleBufferSize not used */
+            CFG_BLE_NUM_GATT_ATTRIBUTES,
+            CFG_BLE_NUM_GATT_SERVICES,
+            CFG_BLE_ATT_VALUE_ARRAY_SIZE,
+            CFG_BLE_NUM_LINK,
+            CFG_BLE_DATA_LENGTH_EXTENSION,
+            CFG_BLE_PREPARE_WRITE_LIST_SIZE,
+            CFG_BLE_MBLOCK_COUNT,
+            CFG_BLE_MAX_ATT_MTU,
+            CFG_BLE_SLAVE_SCA,
+            CFG_BLE_MASTER_SCA,
+            CFG_BLE_LSE_SOURCE,
+            CFG_BLE_MAX_CONN_EVENT_LENGTH,
+            CFG_BLE_HSE_STARTUP_TIME,
+            CFG_BLE_VITERBI_MODE,
+            CFG_BLE_LL_ONLY,
+            0                                   /** TODO Should be read from HW */
+        };
+        /**
+         * Starts the BLE Stack on CPU2
+         */
+        SHCI_C2_BLE_Init(&ble_init_cmd_packet);
+    }
+
+    void transport_init(void)
+    {
+        TL_MM_Config_t tl_mm_config;
+        TL_BLE_InitConf_t tl_ble_config;
+        /* STM32WB offers a System Channel HCI interface for
+           offering system services, with proprietary commands.
+           System Channel must be used as well for starting up
+           BLE service so we need to initialize it. */
+        SHCI_TL_HciInitConf_t shci_init_config;
+
+        /**< Reference table initialization */
+        TL_Init();
+
+        /**< System channel initialization */
+        shci_init_config.p_cmdbuffer = (uint8_t *)&SystemCmdBuffer;
+        shci_init_config.StatusNotCallBack = syscmd_status_not;
+        shci_init(sysevt_received, (void *) &shci_init_config);
+
+        /**< Memory Manager channel initialization */
+        tl_mm_config.p_BleSpareEvtBuffer = BleSpareEvtBuffer;
+        tl_mm_config.p_SystemSpareEvtBuffer = SystemSpareEvtBuffer;
+        tl_mm_config.p_AsynchEvtPool = EvtPool;
+        tl_mm_config.AsynchEvtPoolSize = POOL_SIZE;
+        TL_MM_Init(&tl_mm_config);
+
+        TL_Enable();
+
+        /*  At this stage, we'll need to wait for ready event,
+         *  passed thru TL_SYS_EvtReceived */
+        if (!sysevt_wait()) {
+            /* ERROR booting WB controler */
+            return;
+        }
+
+        /**< BLE channel initialization */
+        tl_ble_config.p_cmdbuffer = (uint8_t *)&BleCmdBuffer;
+        tl_ble_config.p_AclDataBuffer = HciAclDataBuffer;
+        tl_ble_config.IoBusEvtCallBack = evt_received;
+        tl_ble_config.IoBusAclDataTxAck = acl_data_ack;
+        TL_BLE_Init((void *)&tl_ble_config);
+
+        /* Now start BLE service on firmware side, using Vendor specific
+         * command on the System Channel
+         */
+        stm32wb_start_ble();
+
+    }
+
+    int bt_ipm_ble_init(void)
+    {
+        uint16_t       opcode;
+        static uint8_t randCnt;
+#if 0
```

Comment:
Better remove this code for better readibility

Reply:
please remove this code that comes from mbed.. actually looks like the whole function can be removed

Reply:
now, removed as we are sure it is not needed anymore